http/rest-client/src/main/java/io/quarkus/ts/http/restclient/RestInterface.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.ts.http.restclient;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+import io.quarkus.ts.http.restclient.xml.Book;
+
+@RegisterRestClient
+@Path("/book")
+@RegisterClientHeaders
```

Comment:
Just a question: does this annotation do anything if there are no headers specified by a `org.eclipse.microprofile.rest.client.propagateHeaders` property? Based on the way we use it, I assume it by default propagates all incoming headers, but the documentation does not seem entirely clear to me.